spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationSettingsUtil.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.spring;
+
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.spring.ArmeriaSettings;
+
+/**
+ * A utility class which is used to configure a {@link Server} about {@link ArmeriaSettings}.
+ */
+public final class ArmeriaConfigurationSettingsUtil {
+
+    /**
+     * Configures the {@link ServerBuilder} using the specified {@link ArmeriaSettings}.
+     */
+    public static void configureSettings(ServerBuilder server, ArmeriaSettings settings) {
```

Comment:
This method will be invoked only once when a server is started. That means we don't need to seriously consider the performance of this method.
How about setting up these properties using reflection for maintainability.

Reply:
I agree that reflection has more convenient to maintain. And I guess `DataSize` also convenient to users for readability. However, as you know, spring-boot1 doesn't have `DataSize` and so we have to use `String` and `parseDataSize`. Therefore, it was hard to distinguish a `String` field is `DataSize` in reflection.